lib/flow/VaporField.cpp

```diff
@@ -1,37 +1,165 @@
-#include <sstream>
-
 #include "vapor/VaporField.h"
 #include "vapor/ConstantGrid.h"
 #include "GrownGrid.h"
 
+
 using namespace flow;
 
-// Constructor
+
+//
+// Class GridKey
+//
+void GridKey::Reset( uint32_t ts, int32_t ref, int32_t comp, std::string var,
+                     const std::vector<double>& min,
+                     const std::vector<double>& max,
+                     float dz )
+{
+    // See the header file for information stored at each offset.
+    std::memcpy( buf.data(),      &ts,        4 );
```

Comment:
Using sizeof() (or a C++ equivalent) would be preferable to hard-coding the data element size

Reply:
I actually think this is a case hard coded values are more suitable, because there are consecutive memcpy() called with offset values determined by the size value from a previous memcpy() function call.